src/meta/src/manager/catalog/fragment.rs

```diff
@@ -842,18 +924,26 @@ where
     pub async fn get_downstream_chain_fragments(
         &self,
         table_id: TableId,
-    ) -> MetaResult<Vec<Fragment>> {
+    ) -> MetaResult<Vec<(DispatchStrategy, Fragment)>> {
```

Comment:
We also need the output indices in the dispatchers (then map it with column index mapping), so `Fragment` is not sufficient here.